### PR TITLE
BaseActiveRecord now triggers event EVENT_AFTER_REFRESH after a record is refreshed

### LINKED ISSUE #9604: Feature Request : REFRESH event for ActiveRecord
Hi,

**use case** : a behavior is attached to an ActiveRecord model and registers to all apropriates events (AFTER_FIND, AFTER_INSERT, AFTER_UPDATE) so to be able to maintain its internal state aligned with the owner model attributes. 

**the problem** : no event is fired when [refresh()](http://www.yiiframework.com/doc-2.0/yii-db-baseactiverecord.html#refresh%28%29-detail) is called and the behavior is not notified, it can't maintain a consistent internal state.

One solution could be to fire an EVENT_REFRESH when the AciveRecord is refreshed.

thank you

### PR DESCRIPTION
**Pull request by raoul2000**
#9604 : `yii\db\BaseActiveRecord` now triggers event `EVENT_AFTER_REFRESH` after a record is refreshed

Closes #9604

### CODE REVIEW

File: docs/guide/db-active-record.md [hunk 1]
```diff
@@ -570,6 +570,11 @@ life cycle will happen:
 > - [[yii\db\ActiveRecord::updateCounters()]] 
 > - [[yii\db\ActiveRecord::updateAllCounters()]] 
 
+### Refreshing Data Life Cycle <span id="refreshing-data-life-cycle"></span>
+
+When calling [[yii\db\ActiveRecord::refresh()|refresh()]] to refresh an Active Record instance, the event
```
Comment (samdark) [inline docs/guide/db-active-record.md]: the event EVENT_AFTER_REFRESH → EVENT_AFTER_REFRESH event

File: framework/CHANGELOG.md [hunk 2]
```diff
@@ -3,7 +3,7 @@ Yii Framework 2 Change Log
 
 2.0.8 under development
 -----------------------
-
+- Enh #9604: `yii\db\BaseActiveRecord` now triggers event `EVENT_AFTER_REFRESH` after a record is refreshed (raoul2000)
```
Comment (samdark) [inline framework/CHANGELOG.md]: event `EVENT_AFTER_REFRESH` → `EVENT_AFTER_REFRESH` event

File: framework/db/BaseActiveRecord.php [hunk 3]
```diff
@@ -951,6 +964,9 @@ public function afterDelete()
 
     /**
      * Repopulates this active record with the latest data.
+     * 
+     * If the refresh is successful, the event named [[EVENT_AFTER_DELETE]] will be raised by the corresponding methods.
```
Comment (samdark) [inline framework/db/BaseActiveRecord.php]: EVENT_AFTER_DELETE → EVENT_AFTER_REFRESH

**Comment by cebe** [comment]
also needs `@since 2.0.8` phpdoc annotations. @raoul2000 do you have time to finish this?

**Comment by raoul2000** [comment]
Being a git newby I hope the PR is ok ...

File: docs/guide/db-active-record.md (hunk 1 above)
Comment (samdark) [inline docs/guide/db-active-record.md]: This should be fixed.

**Comment by SilverFire** [comment]
:+1:

File: framework/db/BaseActiveRecord.php [hunk 4]
```diff
@@ -951,6 +966,9 @@ public function afterDelete()
 
     /**
      * Repopulates this active record with the latest data.
+     *
+     * If the refresh is successful, the event named [[EVENT_AFTER_DELETE]] will be raised by the corresponding methods.
```
Comment (cebe) [inline framework/db/BaseActiveRecord.php]: EVENT_AFTER_REFRESH, not delete :)

**Comment by cebe** [comment]
Merged with small adjustments, thank you!

**Pull request closed without merging**